File: src/main/java/aurocosh/divinefavor/common/item/contract/ItemFavorContract.java
package aurocosh.divinefavor.common.item.contract;

import aurocosh.divinefavor.DivineFavor;
import aurocosh.divinefavor.common.config.items.Contract;
import net.minecraft.client.resources.I18n;
import net.minecraft.client.util.ITooltipFlag;
import net.minecraft.item.EnumRarity;
import net.minecraft.item.ItemStack;
import net.minecraft.world.World;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

import javax.annotation.Nullable;
import java.util.List;

public class ItemFavorContract extends ItemContract {
    private final int regen;
    private final int min;
    private final int max;
    private final boolean informActivity;

    public ItemFavorContract(String name, String texturePath, Contract contract) {
        this(name, texturePath, contract.regen, contract.minimum, contract.maximum, contract.informActivity);
    }

    public ItemFavorContract(String name, String texturePath, int regen, int min, int max, boolean informActivity) {
        super(name, texturePath);
        this.regen = regen;
        this.min = min;
        this.max = max;
        this.informActivity = informActivity;
        setMaxStackSize(1);
        setCreativeTab(DivineFavor.TAB_MAIN);
    }

    public int getRegen() {
        return regen;
    }

    public int getMin() {
        return min;
    }

    public int getMax() {
        return max;
    }

    public boolean isInformActivity() {
        return informActivity;
    }

    @Override
    public EnumRarity getRarity(ItemStack stack) {
        return EnumRarity.RARE;
    }

    @SideOnly(Side.CLIENT)
    @Override
    public void addInformation(ItemStack stack, @Nullable World world, List<String> tooltip, ITooltipFlag flag) {
        super.addInformation(stack, world, tooltip, flag);
        if (regen != 0)
            tooltip.add(I18n.format("item.divinefavor:contract.regen", regen));
        if (min != 0)
            tooltip.add(I18n.format("item.divinefavor:contract.min", min));
        if (max != 0)
            tooltip.add(I18n.format("item.divinefavor:contract.max", max));
        if (informActivity)
            tooltip.add(I18n.format("item.divinefavor:contract.informActivity"));
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/item/contract/ItemFavorContract.kt
package aurocosh.divinefavor.common.item.contract

import aurocosh.divinefavor.DivineFavor
import aurocosh.divinefavor.common.config.entries.items.Contract
import net.minecraft.client.resources.I18n
import net.minecraft.client.util.ITooltipFlag
import net.minecraft.item.EnumRarity
import net.minecraft.item.ItemStack
import net.minecraft.world.World
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

class ItemFavorContract(name: String, texturePath: String, val regen: Int, val min: Int, val max: Int, val isInformActivity: Boolean) : ItemContract(name, texturePath) {

    constructor(name: String, texturePath: String, contract: Contract) : this(name, texturePath, contract.regen, contract.minimum, contract.maximum, contract.informActivity) {}

    init {
        setMaxStackSize(1)
        creativeTab = DivineFavor.TAB_MAIN
    }

    override fun getRarity(stack: ItemStack): EnumRarity {
        return EnumRarity.RARE
    }

    @SideOnly(Side.CLIENT)
    override fun addInformation(stack: ItemStack, world: World?, tooltip: MutableList<String>, flag: ITooltipFlag) {
        super.addInformation(stack, world, tooltip, flag)
        if (regen != 0)
            tooltip.add(I18n.format("item.divinefavor:contract.regen", regen))
        if (min != 0)
            tooltip.add(I18n.format("item.divinefavor:contract.min", min))
        if (max != 0)
            tooltip.add(I18n.format("item.divinefavor:contract.max", max))
        if (isInformActivity)
            tooltip.add(I18n.format("item.divinefavor:contract.informActivity"))
    }
}
